### DISCUSSION TITLE
[6 luglio 2023] Calcolabilità

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/automi-calcolabilita-complessita#14] [Calcolabilità]
![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/69389461/d224d435-3f2f-45f8-93cb-145926d96db7)

**Comment by Exyss** [comment]
Ho preferito dimostrare direttamente che il linguaggio sia irriconoscibile (che ovviamente implica che sia indecidibile)

![image](https://github.com/user-attachments/assets/5a57bf70-c8be-4e5b-8709-8b699e9f1741)

----

La dimostrazione richiesta nel secondo punto è presente sul libro di testo

**Reply by 18xChris** [reply]
Per il secondo punto propongo la soluzione presente sugli appunti di [CasuFrost](https://github.com/CasuFrost/University_notes)
![image](https://github.com/user-attachments/assets/71436189-07d4-4194-abc9-6f50e1f19410)
![image](https://github.com/user-attachments/assets/9da744d3-30a9-459f-9932-7a6d2ea7fc25)

**Comment by daromacarlo** [comment]
<img width="1570" height="638" alt="immagine" src="https://github.com/user-attachments/assets/33554019-638e-4ed0-aecd-b8a2a87ee34d" />

**Reply by AglaiaNorza** [reply]
sadly mi sa che la tua soluzione non funziona, perché se non erro hai:
- se M(w) = ACC, M' accetta quindi L(M') = $\Sigma^*$ che è decidibile, ovvero D accetta e S rifiuta
- se M(w) = REJ, M' accetta comunque, quindi L(M') = $\Sigma^*$ che è decidibile, ovvero D accetta e S rifiuta
- se M(w) = LOOP, M' va in loop quindi L(M') = $\emptyset$  che è decidibile, ovvero D accetta e S rifiuta

quindi essenzialmente hai una TM che rifiuta sempre

(anche se avessi fatto "altrimenti rifiiuta" in M', avresti avuto
- se M(w) = REJ, M' rifiuta, quindi L(M') = $\emptyset$ che è decidibile, ovvero D accetta e S rifiuta
)

**Reply by daromacarlo** [reply]
Vero, l'ho rifatto, spero funzioni. **Grazie per l'aiuto**
<img width="669" height="258" alt="Immagine 2026-02-05 001115" src="https://github.com/user-attachments/assets/94be0ca6-e009-429d-aa40-c295a005b585" />
<img width="668" height="344" alt="Immagine 2026-02-05 001128" src="https://github.com/user-attachments/assets/d008b07e-ced6-400d-b232-c6a88c732137" />
<img width="668" height="79" alt="Immagine 2026-02-05 001139" src="https://github.com/user-attachments/assets/d2aa74e2-c494-4c66-b903-ab50ed2b75f1" />
